modules/tinymce/src/themes/silver/main/ts/ui/button/ButtonSlices.ts

```diff
@@ -8,18 +8,11 @@
 import { Behaviour, Replacing, SimpleOrSketchSpec } from '@ephox/alloy';
 
 import { UiFactoryBackstageProviders } from '../../backstage/Backstage';
-import { get as getIcon, IconProvider } from '../icons/Icons';
+import { get as getIcon, IconProvider, render as renderIconElement } from '../icons/Icons';
 import { ToolbarButtonClasses } from '../toolbar/button/ButtonClasses';
 
-const renderIcon = (iconHtml, behaviours): SimpleOrSketchSpec =>
-  ({
-    dom: {
-      tag: 'span',
-      innerHtml: iconHtml,
-      classes: [ ToolbarButtonClasses.Icon, ToolbarButtonClasses.IconWrap ]
-    },
-    ...behaviours
-  });
+const renderIcon = (iconHtml: string, behaviours: Partial<SimpleOrSketchSpec>): SimpleOrSketchSpec =>
+  renderIconElement('span', iconHtml, [ ToolbarButtonClasses.Icon, ToolbarButtonClasses.IconWrap ], behaviours);
```

Comment:
I know this was previously named this way, but am I right in thinking that `behaviours` could really be any partial of the `ComponentSpec`? That means you could pass in `{ events: ...something }` and it would override the `AlloyEvents` you've specified in the `render` in `Icons.ts`.

If this is only ever actual behaviours, maybe it should be `AlloyBehaviourRecord` and not be spread into the object in `Icons.ts`.

This may not be something we care about (or I may be wrong) I just wanted to raise it

Reply:
Hmm I know what you mean (after reading it 5x hehe) but I don't think I have knowledge of the codebase to answer. Maybe @lnewson ?

Reply:
Looking at the code, only `renderIconFromPack` and `renderReplacableIconFromPack` from `ButtonSlices` call `render` from `Icon` passing behaviours. `renderIconFromPack` passes an empty object and `renderReplacableIconFromPack` passes behaviours (no override). That said, I think you are correct, we don't need the spread and can change the type.

Reply:
Yeah, this is what I mentioned in our call the other day. We need to either rename `behaviours` to something more accurate, or make it `Optional<AlloyBehaviourRecord>`. The latter is tricker since it's not always passed, so the `behaviours` should only be in the spec if it's `some`.

Reply:
Change the type to `AlloyBehaviourRecord` and remove the spread.

Reply:
As I mentioned above, it needs to be `Optional` as we shouldn't be passing `undefined` into the spec, so this needs changing please.

- [ ] Fix as per comments above

Reply:
ah right, "we shouldn't be passing undefined" triggered the optional in my head :)